packages/application-shell/src/components/navbar/navbar.js

```diff
@@ -607,149 +606,100 @@ NavBarLayout.propTypes = {
-      servedByProxy: PropTypes.bool.isRequired,
-      useFullRedirectsForLinks: PropTypes.bool.isRequired,
-    }).isRequired,
-    menuVisibilities: PropTypes.objectOf(PropTypes.bool).isRequired,
-    // Injected
-    areProjectExtensionsEnabled: PropTypes.bool.isRequired,
-    location: PropTypes.object.isRequired,
-    isForcedMenuOpen: PropTypes.bool,
-    applicationsMenuQuery: PropTypes.shape({
-      applicationsMenu: PropTypes.shape({
-        navBar: PropTypes.arrayOf(PropTypes.object).isRequired,
-      }),
-    }),
-    projectExtensionsQuery: PropTypes.shape({
-      projectExtension: PropTypes.shape({
-        id: PropTypes.string.isRequired,
-        applications: PropTypes.arrayOf(
-          PropTypes.shape({
-            id: PropTypes.string.isRequired,
-            navbarMenu: PropTypes.object.isRequired,
-          })
-        ).isRequired,
-      }),
-    }),
-  };
-
-  ref = React.createRef();
-
-  render() {
-    const navbarMenu =
-      (this.props.applicationsMenuQuery &&
-        this.props.applicationsMenuQuery.applicationsMenu &&
-        this.props.applicationsMenuQuery.applicationsMenu.navBar) ||
-      [];
-    const customAppsMenu =
-      this.props.projectExtensionsQuery &&
-      this.props.projectExtensionsQuery.projectExtension
-        ? this.props.projectExtensionsQuery.projectExtension.applications.map(
-            app => app.navbarMenu
-          )
-        : [];
-    return (
-      <NavBarLayout ref={this.ref}>
-        <DataMenu
-          rootNode={this.ref.current}
-          data={navbarMenu.concat(customAppsMenu)}
-          isForcedMenuOpen={this.props.isForcedMenuOpen}
-          location={this.props.location}
-          menuVisibilities={this.props.menuVisibilities}
-          applicationLocale={this.props.applicationLocale}
-          projectKey={this.props.projectKey}
-          useFullRedirectsForLinks={
-            this.props.environment.useFullRedirectsForLinks
-          }
-        />
-      </NavBarLayout>
-    );
-  }
-}
-
-const injectMenuToggleState = Component => {
-  const WrappedComponent = props => {
-    const cachedIsForcedMenuOpen = storage.get(
-      STORAGE_KEYS.IS_FORCED_MENU_OPEN
-    );
-    return (
-      <Component
-        {...props}
-        isForcedMenuOpen={
-          typeof cachedIsForcedMenuOpen === 'string'
-            ? cachedIsForcedMenuOpen === 'true'
-            : null
-        }
-      />
-    );
-  };
-  WrappedComponent.displayName = `injectMenuToggleState(${getDisplayName(
-    Component
-  )})`;
-  return WrappedComponent;
-};
-
-export default flowRight(
-  withRouter, // Connect again, to access the `location` object
-  injectFeatureToggle('projectExtensions', 'areProjectExtensionsEnabled'),
-  injectMenuToggleState,
-  withApplicationsMenu({
-    queryName: 'applicationsMenuQuery',
-    skipRemoteQuery: ownProps => !ownProps.environment.servedByProxy,
-    options: ownProps => ({
+export const NavBar = props => {
```

Comment:
We're using several hooks here and the code is much more readable now!

Reply:
I agree and appriciate this. However, I would also argue that we should split updating libs and  refactoring code. So we only change as little as possible in dep updates and keep such changes for other PRs. I am just arguing for this in the future as we had quite a few regressions in the last weeks.

Reply:
Yeah sure, I agree 👌

Just for the record, those refactorings were part of the experimental branch of using the Apollo hooks 2 months ago or something.